doc/Dhcp_Mitigation/DHCP Mitigation.md

```diff
@@ -0,0 +1,216 @@
+
+### Use Cases for DHCP DOS 
+
+
+A flood of DHCP discover packets affects the DHCP relay agent. DHCP relay agent forwards DHCP messages between clients and servers across different VLANs. With  large number of DHCP discover packets the relay agent becomes overwhelmed, leading to various issues:
+
+#### Resource Exhaustion 
+The DHCP relay agent has finite resources such as CPU, memory, and network bandwidth. A flood of DHCP discover packets consumes these resources, causing the relay agent to become overloaded and unable to process legitimate DHCP requests effectively.
+
+#### Service Degradation
+The flood of DHCP discover packets causes service degradation for other clients served by the DHCP relay agent. Legitimate DHCP requests experiences delays or timeouts as the relay agent struggles to handle the excessive volume of traffic.
+
+#### Packet Loss
+ The DHCP relay agent experiences packet loss due to the overwhelming volume of DHCP discover packets. This can result in dropped DHCP requests from legitimate clients, leading to connectivity issues and network downtime.
+
+#### Impact on Network Performance
+The flood of DHCP discover packets impacts the performance of the network infrastructure, including switches, routers, and other devices involved in forwarding DHCP traffic. Congestion caused by the flood can degrade overall network performance and affect the operation of other network services.
+
+
+### Behavior of DHCP DoS Attack
+
+
+![image](../../images/dhcp_mitigation_hld/topo.png)
+
+
+The above figure shows 2 different VLANs configured with 4 hosts. Attacker and Host-1 are in Vlan-20 and DHCP-SERVER and Host-2 reside in Vlan-10.  
+
+As shown in the diagram, the attacker launches an attack on the DHCP server/relay with a flood of DHCP DISCOVER packets. The attacker generates a large number of DHCP DISCOVER packets with spoofed MAC addresses and sends these packets to the DHCP relay. The DHCP relay forwards these packets to the DHCP server. The attack is shown in the following diagram (captured from wireshark during instance of attack).  
+
+
+![image](../../images/dhcp_mitigation_hld/Discover.png)
+
+
+Due to the huge number of DHCP DISCOVER packets, the server is busy in dealing the DHCP DISCOVER packets and unable to respond to other clients on the same VLAN as the attacker, showing a successful DoS Attack. This demonstrates a DOS Attack which overwhelms the DHCP server.
+
+It can be seen in the screenshot below that a legitimate client (PC1) is unable to be serviced by the DHCP server. After the continuous bombardment of DHCP DISCOVER packets, the DORA process fails to complete, indicating a potential Denial of Service (DoS) situation.
+
+
+![image](../../images/dhcp_mitigation_hld/DORA.png)
+
+
+### DHCP DoS Mitigation
+
+
+To prevent possible DHCP DoS attacks in SONiC, we suggest using rate-limiting with Linux Traffic Control (TC). 
```

Comment:
Can existing COPP rate-limiting policy mitigate this issue?
With COPP policy, the packets will not even trapped to CPU when exceeding pre-defined rate-limit.

+ @yaqiangz @kellyyeh to comment

Reply:
Yes, existing COPP can mitigate this issue. But I think advantage of this proposal is that it can configure per interface to narrow down the limit granularity

Reply:
Currently, COPP also resolves rate limiting but not specific focus for per interface. The proposed solution provides changes on per interface.